fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java

```diff
@@ -119,16 +124,108 @@ public int getMaxArity() {
                     }
                 }
                 return membershipCheckFailed(evaluationContext, elementNode, url, strength);
-            } else {
+            }
+            else if (isSyntaxBased(valueSet)) {
+                
+                // Validate against syntax-based value set
+                if (validateAgainstSyntaxBasedValueSet(valueSet, evaluationContext, elementNode, strength)) {
+                    return SINGLETON_TRUE;
+                }
+                return membershipCheckFailed(evaluationContext, elementNode, url, strength);
+            }
+            else {
                 generateIssue(evaluationContext, IssueSeverity.WARNING, IssueType.INCOMPLETE, "Membership check was not performed: value set '" + url + "' is empty or could not be expanded", elementNode.path());
             }
-        } else {
+        }
+        else {
             generateIssue(evaluationContext, IssueSeverity.WARNING, IssueType.NOT_SUPPORTED, "Membership check was not performed: value set '" + url + "' is not supported", elementNode.path());
         }
 
         return SINGLETON_TRUE;
     }
 
+    /**
+     * Determines if the value set is syntax-based.
+     * @param valueSet the value set
+     * @return true or false
+     */
+    private boolean isSyntaxBased(ValueSet valueSet) {
+        String valueSetUrl = valueSet.getUrl() != null ? valueSet.getUrl().getValue().split("\\|")[0] : null;
+        return ALL_LANG_VALUE_SET_URL.equals(valueSetUrl);
+    }
+    
+    /**
+     * Validates the element against the syntax-based value set.
+     * @param valueSet the value set
+     * @param evaluationContext the evaluation context
+     * @param elementNode the element node to validate
+     * @param strength the binding strength
+     * @return true if validation was successful, otherwise false
+     */
+    private boolean validateAgainstSyntaxBasedValueSet(ValueSet valueSet, EvaluationContext evaluationContext, FHIRPathElementNode elementNode, String strength) {
+        Element element = elementNode.element();
+        Code code = null;
+        Coding coding = null;
+        CodeableConcept codeableConcept = null;
+
+        // Determine the system/version/code or CodableConcept to validate
+        if (element.is(Code.class)) {
+            code = element.as(Code.class);
+        }
+        else if (element.is(Coding.class)) {
+            coding = element.as(Coding.class);
+        }
+        else if (element.is(CodeableConcept.class)) {
+            codeableConcept = element.as(CodeableConcept.class);
+        }
+        else if (element.is(Quantity.class)) {
+            Quantity quantity = element.as(Quantity.class);
+            coding = Coding.builder().system(quantity.getSystem()).code(quantity.getCode()).build();
+        }
+        else {
+            code = element.is(FHIR_STRING) ? Code.of(element.as(FHIR_STRING).getValue()) : Code.of(element.as(Uri.class).getValue());
```

Comment:
seems a bit odd not to handle the `element.is(Uri.class)` in the if/else block with the others, but I suppose its fine

Reply:
I kept the if/else same logic used in the apply(...) method for expanded value sets (around line 121) to be consistent.  I can change both places if there's a way that would be better.